### PR TITLE
Security updates to APIM and AppService

### PR DESCRIPTION
**Pull request by pamelafox**
These are two security updates to the /core modules based on the Microsoft Security recommendations. (See https://github.com/Azure/azure-dev/issues/1640)

These both seem like options that won't negatively affect the apps.

### CODE REVIEW

File: templates/common/infra/bicep/core/gateway/apim.bicep [hunk 1]
```diff
@@ -37,6 +37,23 @@ resource apimService 'Microsoft.ApiManagement/service@2021-08-01' = {
...
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_RSA_WITH_AES_256_CBC_SHA': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_RSA_WITH_AES_128_CBC_SHA': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TripleDes168': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls10': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls11': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Ssl30': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Tls10': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Tls11': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Ssl30': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Protocols.Server.Http2': 'false'
```
Comment (ellismg) [inline templates/common/infra/bicep/core/gateway/apim.bicep:55]: I was sort of surprised to see the `Http2` rule here, but it looks like this was the default behavior anyway: 

> Http2 setting's default value is False.

(from: https://learn.microsoft.com/en-us/azure/templates/microsoft.apimanagement/service?pivots=deployment-language-bicep)

**Review by ellismg — APPROVED** [review]

File: templates/common/infra/bicep/core/gateway/apim.bicep (hunk 1 above)
Comment (pamelafox) [reply templates/common/infra/bicep/core/gateway/apim.bicep:55]: Ah I could remove it in that case, I think we generally don't want to replicate defaults.

**Review by ellismg — APPROVED** [review]
LGTM

File: templates/common/infra/bicep/core/gateway/apim.bicep (hunk 1 above)
Comment (ellismg) [reply templates/common/infra/bicep/core/gateway/apim.bicep]: Thanks for the updates, @pamelafox!

File: templates/common/infra/bicep/core/gateway/apim.bicep [hunk 2]
```diff
@@ -37,6 +37,22 @@ resource apimService 'Microsoft.ApiManagement/service@2021-08-01' = {
...
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_RSA_WITH_AES_256_CBC_SHA': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_RSA_WITH_AES_128_CBC_SHA': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TripleDes168': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls10': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls11': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Ssl30': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Tls10': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Tls11': 'false'
+      'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Ssl30': 'false'
+    }
```
Comment (wbreza) [inline templates/common/infra/bicep/core/gateway/apim.bicep:55]: FYI, receiving error during ARM deployment with the following.  Looks like custom properties are not supported when SKU == `Consumption`

`json
{
    "status": "Failed",
    "error": {
        "code": "NotSupported",
        "message": "'Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_RSA_WITH_AES_128_GCM_SHA256,Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_RSA_WITH_AES_256_CBC_SHA256,Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_RSA_WITH_AES_128_CBC_SHA256,Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_RSA_WITH_AES_256_CBC_SHA,Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TLS_RSA_WITH_AES_128_CBC_SHA,Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TripleDes168,Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Ssl30' customProperties are not supported in SkuType Consumption."
    }
}
`

Comment (pamelafox) [reply templates/common/infra/bicep/core/gateway/apim.bicep:55]: The fix in your PR looks good!

**Pull request merged**